### PR TITLE
Improve advice around fs.inotify.max_user_watches

### LINKED ISSUE #235: Advise Linux users how to load in the ENOSPC config
Context: https://github.com/Microsoft/vscode/issues/4252

http://code.visualstudio.com/Docs/supporting/faq#_linux-faq

```bash
sudo sysctl -p
```

### PR DESCRIPTION
**Pull request by Tyriar**
Fixes #235, #236

### CODE REVIEW

**Comment by Tyriar** [comment]
@gregvanl let me know if wording looks good.

File: docs/supporting/faq.md [hunk 1]
```diff
@@ -146,13 +146,13 @@ When you see this error, it indicates that the VS Code file watcher is running o
...
 
-The limit can be increased to its maximum by editing `/etc/sysctl.conf` and add this line to the end of the file:
+The limit can be increased to its maximum by editing `/etc/sysctl.conf` and adding this line to the end of the file:
 
 ```
 fs.inotify.max_user_watches=524288
 ```
 
-The new value can then be loaded in by running `sudo sysctl -p`.
+The new value can then be loaded in by running `sudo sysctl -p`. Note that ArchLinux works a little different, [view this page for advice](https://github.com/guard/listen/wiki/Increasing-the-amount-of-inotify-watchers).
```
Comment (gregvanl) [inline docs/supporting/faq.md]: "Note that [ArchLinux](https://www.archlinux.org) works a little differently, "

Comment (Tyriar) [inline docs/supporting/faq.md]: :apple: works a little different

File: docs/supporting/faq.md [hunk 2]
```diff
@@ -140,11 +140,19 @@ sudo apt-get install gvfs-bin
...
 ```
-fs.inotify.max_user_watches=16384
+
+The limit can be increased to its maximum by editing `/etc/sysctl.conf` and add this line to the end of the file:
+
 ```
+fs.inotify.max_user_watches=524288
+```
+
+While 524288 is the maximum number of files that can be watched, if you're in an environment that is particularly memory constrained you may wish to lower the number. Each file watch [takes up 540 bytes (32-bit) or ~1kB (64-bit)](http://stackoverflow.com/a/7091897/1156119), so assuming that all 524288 watches are consumed that results in an upperbound of around 256MB (32-bit) or 512MB (64-bit).
```
Comment (gregvanl) [inline docs/supporting/faq.md]: "constrained, you may"

**Comment by gregvanl** [comment]
Thanks Dan for writing this up.  I just add a few suggestions.

**Comment by Tyriar** [comment]
Cheers :)

**Comment by gregvanl** [comment]
Do you think we should link to https://www.archlinux.org?

**Comment by Tyriar** [comment]
May as well, I notice it's also stylized as "Arch Linux". I'll fix that up.

**Comment by gregvanl** [comment]
Cool, nice to link people directly instead of them having to type in search if they aren't familiar with a technology.

**Pull request merged**